docs/develop/go/test-suites.mdx

```diff
@@ -160,6 +161,103 @@ When an Activity sends a Heartbeat, be sure that you can see the Heartbeats in y
+                SampleInput{},
+                workflow.NexusOperationOptions{},
+        ).Return(
+                &nexus.HandlerStartOperationResultSync[SampleOutput]{
+                        Value: SampleOutput{},
+                },
+                nil, // error if you want to simulate an error in the ExecuteOperation call
+        )
+        // You can also add a delay to return the mock values by calling After().
+        // Eg: s.env.OnNexusOperation(...).Return(...).After(1*time.Second)
+
+        s.env.ExecuteWorkflow(SimpleWorkflow, "test_nexus_operation")
+
+        s.True(s.env.IsWorkflowCompleted())
+        s.NoError(s.env.GetWorkflowError())
+}
+```
+
+The example above simulates a call to a Nexus synchronous operation, indicated by the returned value
+type `*nexus.HandlerStartOperationResultSync[T]`. Since `OnNexusOperation` needs to know the
+operation's name, input and output, `nexus.NewOperationReference` creates an operation reference
+that represents the operation without knowing the implementation (basically, it represents the
+signature of the operation). You may also use the operation itself instead of creating the operation
+reference if you have it available.
+
+Let's take an example of simulating a call to an Nexus asynchronous operation.
+
+```go
+func (s *UnitTestSuite) Test_SimpleWorkflow_NexusAsyncOperation() {
+        s.env.OnNexusOperation(
+                "sample-service",
+                nexus.NewOperationReference[SampleInput, SampleOutput]("sample-operation"),
+                SampleInput{},
+                workflow.NexusOperationOptions{},
+        ).Return(
+                &nexus.HandlerStartOperationResultAsync{
+                        OperationID: "sample-operation-id",
+                },
+                nil, // error if you want to simulate an error in the ExecuteOperation call
+        )
+        err := env.RegisterNexusAsyncOperationCompletion(
+                "sample-service",
+                "sample-operation",
+                "sample-operation-id", // must match the OperationID above
+                SampleOutput{},
+                nil,            // error if you want to simulation an error in the operation
+                2*time.Second,  // delay to simulate how long the operation takes after it starts
+        )
+        s.NoError(err)
+        s.env.ExecuteWorkflow(SimpleWorkflow, "test_nexus_operation")
+
+        s.True(s.env.IsWorkflowCompleted())
+        s.NoError(s.env.GetWorkflowError())
+}
+```
+
+In the example above, the `OperationID` in the `*nexus.HandlerStartOperationResultAsync` object can
+be any string of your choice, and it is used in the `RegisterNexusAsyncOperationCompletion` call to
+register the result of the corresponding asynchronous operation. With this, you can mock multiple
+asynchronous Nexus operations and register the results for each of them in the same workflow.
+
+If you need to run some custom logic for the Nexus operation, you can do as follows.
```

Comment:
I'd like to see in-text what the custom logic is, and how that differs from the previous examples.

Reply:
By custom logic, I mean any logic the user might want. Let try to be more clear.